app/models/day.rb

```diff
@@ -3,20 +3,12 @@ class Day < ActiveRecord::Base
   validates :count, presence: true
   validates :date, presence: true, uniqueness: true
 
-  def self.to_json_object
-    year_range = (minimum(:date).year..maximum(:date).year)
-    year_range.each_with_object({}) do |year, json_object|
-      json_object[year] = days_within_year(year).map do |day|
-        {
-          date: day.date,
-          count: day.count,
-        }
-      end
-    end.to_json
-  end
-
-  def self.days_within_year(year)
-    where(date: Date.new(year, 1, 1)..Date.new(year, 12, 31)).order(:date)
+  def calculate_moving_average_days(n) # this should be done in sql
+    range_start = (date - n.days)
+    return nil unless Day.where("date < ?", range_start).present?
```

Comment:
Style/StringLiterals: Prefer single-quoted strings when you don't need string interpolation or special symbols.